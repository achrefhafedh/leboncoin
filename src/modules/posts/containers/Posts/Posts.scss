@import '../../../../assets/colors.scss';

.posts {
  background: $blue;
  margin: 15px 0px;
  color: $white;
  padding: 20px;
}

.empty {
  text-align: center;
  font-size: 1.3rem;
  padding: 30px;
}
