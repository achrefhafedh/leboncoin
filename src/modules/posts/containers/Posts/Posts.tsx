import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getModuleState } from 'redux-register-module';

import { Store } from 'store';
import { PostType } from '../../types';
import { postsHeaderLayoutHOC } from '../../hoc/postsHeaderLayout';

type OwnProps = {};

type StateProps = {
  posts: PostType[];
  userType: string;
};

type DispatchProps = {};

type PostsProps = StateProps & DispatchProps & OwnProps;

type PostsState = {
  posts: PostType[];
  userType: string;
};
class PostsContainer extends Component<PostsProps, PostsState> {
  constructor(props: PostsProps) {
    super(props);
    this.state = {
      posts: [],
      userType: '',
    };
  }
  componentDidMount() {
    const { posts, userType } = this.props;
    if (posts && userType) {
      this.setState(() => ({
        posts,
        userType,
      }));
    }
  }
  renderContent = () => {
    const { posts, userType } = this.props;
    if (!posts || !userType) return null;
    return (
      <>
        {posts.length === 0 && <div className="empty">Empty Posts</div>}
        {posts.map((post, key) => (
          <div className="post" key={key}>
            {post.content}
          </div>
        ))}
      </>
    );
  };
  render() {
    return this.renderContent();
  }
}

const mapStateToProps = (state: Store, onProps: OwnProps): StateProps => {
  return {
    userType: getModuleState('login', state).user.type,
    posts: getModuleState('posts', state).posts,
  };
};

const PostsConnected = connect(
  mapStateToProps,
  {}
)(PostsContainer);

export const Posts = postsHeaderLayoutHOC(<PostsConnected />);
