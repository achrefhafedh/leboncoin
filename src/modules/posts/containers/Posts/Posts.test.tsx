import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Posts } from './Posts';

configure({ adapter: new Adapter() });

describe('render without crashing', () => {
  beforeEach(() => {});

  it('render without crashing', () => {
    const auth = shallow(<Posts />);
    expect(auth).toMatchSnapshot();
  });
});
