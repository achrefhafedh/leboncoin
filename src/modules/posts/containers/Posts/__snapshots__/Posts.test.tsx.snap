// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render without crashing render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderLayout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div>
          <Header />
        </div>,
        <div>
          <Connect(PostsContainer) />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Header />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(PostsContainer) />,
          "className": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div>
            <Header />
          </div>,
          <div>
            <Connect(PostsContainer) />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Header />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(PostsContainer) />,
            "className": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
